imap/squatter.c

```diff
@@ -206,29 +215,46 @@ static int should_index(const char *name)
                extname, error_message(r));
 
         free(extname);
-        return 0;
+        ret = 0;
+        goto done;
     }
 
     // skip remote or not-real mailboxes
     if (mbentry->mbtype & (MBTYPE_REMOTE|MBTYPE_DELETED|MBTYPE_INTERMEDIATE)) {
-        mboxlist_entry_free(&mbentry);
-        return 0;
+        ret = 0;
+        goto done;
     }
 
     // skip email submissions
     if (mboxname_issubmissionmailbox(mbentry->name, mbentry->mbtype)) {
-        mboxlist_entry_free(&mbentry);
-        return 0;
+        ret = 0;
+        goto done;
     }
 
     // skip COLLECTION mailboxes (just files)
     if (mbentry->mbtype & MBTYPE_COLLECTION) {
-        mboxlist_entry_free(&mbentry);
-        return 0;
+        ret = 0;
+        goto done;
+    }
+
+    // skip listed domains
+    if (mbname_domain(mbname) && skip_domains &&
+        strarray_find(skip_domains, mbname_domain(mbname), 0) > 0) {
```

Comment:
0 is a legit index, so this should read `>= 0`

Reply:
yes, you're right.  Good point!